### PR TITLE
Add tests with pin-source parameter in PKCS#11 URI

### PR DESCRIPTION
**Pull request by sarroutbi**
Resolves: #383

#### Description
Add tests with pin-source parameter in PKCS#11 URI.

#### Checklist
- [x] Test suite updated with functionality tests

#### Reviewer's checklist:
- [x] Any issues marked for closing are addressed
- [x] There is a test suite reasonably covering new functionality or modifications
- [x] This feature/change has adequate documentation added
- [x] Code conform to coding style that today cannot yet be enforced via the check style test
- [x] Commits have short titles and sensible commit messages
- [x] Coverity Scan has run if needed (code PR) and no new defects were found

### CODE REVIEW

**Comment by sarroutbi** [comment]
Please, hold this PR until I debug issues in Debian.

**Comment by sarroutbi** [comment]
I will increase the test timeouts, as I am seeing wider times than 30 seconds in some tests:
```bash
...
 5/33 pkcs11-provider:softhsm / certs            OK              33.85s
 6/33 pkcs11-provider:softokn / basic            OK              36.04s
 7/33 pkcs11-provider:softhsm / pubkey           OK              48.54s
 8/33 pkcs11-provider:softhsm / ecc              OK              70.15s
 9/33 pkcs11-provider:softhsm / edwards          OK              69.19s
10/33 pkcs11-provider:softhsm / basic            OK             133.68s
...
32/33 pkcs11-provider:softokn / uri              OK              66.53s
33/33 pkcs11-provider:softhsm / uri              OK             190.58s
```

File: tests/meson.build [hunk 1]
```diff
@@ -151,6 +151,7 @@ foreach t, extra_args : tests
       depends: [softoken_conf, softhsm_conf],
       env: test_env,
       is_parallel: is_parallel,
+      timeout: 300,
```
Comment (simo5) [inline tests/meson.build:154]: Why a 5 minutes timeout?

Comment (simo5) [reply tests/meson.build:154]: Do you know what is causing it to take so much time on your system?
On mine I can complete tests in a few seconds once setup is completed ...
The longest test on my laptop takes 5 seconds, most are sub-second

**Comment by simo5** [comment]
(FYI ignore whatever covscan result we get, I am running it only to check it works, but it may fail, I need a datapoint on a real PR to adjust something in it)

File: tests/meson.build (hunk 1 above)
Comment (sarroutbi) [reply tests/meson.build:154]: > Do you know what is causing it to take so much time on your system? On mine I can complete tests in a few seconds once setup is completed ... The longest test on my laptop takes 5 seconds, most are sub-second

I always perform clean compile and test. If compilation is cached, I guess it should not be necessary

Comment (simo5) [reply tests/meson.build]: Is this a particularly slow system?

This is my timing:
```
$ time (git clean -f -x -d -f; make; make check)
[..]
Ok:                 33  
Expected Fail:      0   
Fail:               0   
Unexpected Pass:    0   
Skipped:            0   
Timeout:            0   

Full log written to /home/remote/ssorce/devel/git/pkcs11-provider/builddir/meson-logs/testlog.txt

real	0m12.833s
user	0m14.241s
sys	0m2.443s
```

About 15s total including compilation from a clean tree

Comment (sarroutbi) [reply tests/meson.build]: > Is this a particularly slow system?

No, it isn´t:
```bash
$ lscpu
Architecture:             x86_64
  CPU op-mode(s):         32-bit, 64-bit
  Address sizes:          39 bits physical, 48 bits virtual
  Byte Order:             Little Endian
CPU(s):                   8
  On-line CPU(s) list:    0-7
Vendor ID:                GenuineIntel
  Model name:             Intel(R) Core(TM) i7-10610U CPU @ 1.80GHz
...
```
> 
> This is my timing:
> 
> ```
> $ time (git clean -f -x -d -f; make; make check)
> [..]
> Ok:                 33  
> Expected Fail:      0   
> Fail:               0   
> Unexpected Pass:    0   
> Skipped:            0   
> Timeout:            0   
> 
> Full log written to /home/remote/ssorce/devel/git/pkcs11-provider/builddir/meson-logs/testlog.txt
> 
> real	0m12.833s
> user	0m14.241s
> sys	0m2.443s
> ```
> 
> About 15s total including compilation from a clean tree

I just removed everything and, although it is taking longer than yours, tests are passing correctly, 10 secs the longer one. My machine should be running some weird/heavy process ... I already removed the timeout change.

**Pull request merged**